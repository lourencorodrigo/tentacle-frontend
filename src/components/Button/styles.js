import styled from 'styled-components';

import { rem } from '../../styles/tools';

export const Wrapper = styled.button`
  color: #fff;
  font-family: Avenir;
  font-weight: bold;
  font-size: ${rem(16)};
  background: ${props => props.theme.primary};
  border: 0;
  padding: ${rem(10)} ${rem(16)};
  width: 100%;
  display: inline-block;
  border-radius: ${rem(4)};
  user-select: none;
  border-radius: ${rem(6)};
  letter-spacing: ${rem(1.5)};
  transition: background 0.3s ease-in-out;

  &:focus,
  &:hover {
    cursor: pointer;
    background: #6136c1;
    text-decoration: none;
    outline: 0;
  }

  &:disabled {
    background: ${props => props.theme.gray48};
    cursor: default;
  }
`;
