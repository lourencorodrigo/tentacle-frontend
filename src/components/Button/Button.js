import React from 'react';
import PropTypes from 'prop-types';

import { Wrapper } from './styles';

const Button = ({ children, type, disabled }) => (
  <Wrapper type={type} disabled={disabled}>
    {children}
  </Wrapper>
);

Button.propTypes = {
  children: PropTypes.string.isRequired,
  type: PropTypes.string,
  disabled: PropTypes.bool
};

export default Button;
