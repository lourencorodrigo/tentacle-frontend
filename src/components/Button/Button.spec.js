import React from 'react';
import ReactDOM from 'react-dom';
import ReactTestRenderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Button from './';

describe('button component', () => {
  let wrapper;
  const button = <Button>Test</Button>;

  beforeEach(() => {
    wrapper = mount(button);
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(button, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should compare the text.', () => {
    expect(
      wrapper
        .find('button')
        .render()
        .text()
    ).toEqual('Test');
  });

  it('should compare the component with a snapshot', () => {
    const component = ReactTestRenderer.create(button);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
