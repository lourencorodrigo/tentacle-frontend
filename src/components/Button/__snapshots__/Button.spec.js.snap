// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component should compare the component with a snapshot 1`] = `
<button
  className="sc-bdVaJa cdOaBd"
>
  Test
</button>
`;
